.github/MIGRATION.md

```diff
@@ -14,6 +20,131 @@
+
+```ts
+import { STACKS_MAINNET } from '@stacks/network';
+import { STACKS_TESTNET } from '@stacks/network';
+import { STACKS_MOCKNET } from '@stacks/network';
+import { STACKS_DEVNET } from '@stacks/network';
+```
+
+### StacksNodeApi
+
+The new `StacksNodeApi` class lets you interact with a Stacks node or API.
+
+<!-- todo: will be renamed to Client in a followup PR -->
+
+```ts
+import { StacksNodeApi } from '@stacks/transactions';
+
+const api = new StacksNodeApi();
+await api.broadcastTx(txHex);
+```
+
+### StacksNetwork to StacksNodeApi
+
+Stacks network objects are now exported by the `@stacks/common` package.
+They are used to specify network settings for other functions and don't require instantiation (like the `@stacks/network` approach did).
+
+```ts
+import { STACKS_MAINNET } from '@stacks/transactions';
+```
+
+After importing the network object (e.g. `STACKS_MAINNET` here), you can use it in other functions like so:
+
+```ts
+// todo: update more functions, show example
+```
+
+For easing the transition, the functions which depended on a network instance now accept an `api` parameter.
+The `api` parameter can be an instance of `StacksNodeApi` or any object containing a `url` and `fetch` property.
+
+- The `url` property should be a string containing the base URL of the Stacks node you want to use.
+- The `fetch` property can be any (fetch)[https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API] compatible function.
+
+The following diffs show examples of how to migrate to the new pattern.
+
+```diff
+import { makeSTXTokenTransfer } from '@stacks/transactions';
+
+- import { StacksTestnet } from '@stacks/network';
++ import { STACKS_TESTNET } from '@stacks/network';
+
+const transaction = await makeSTXTokenTransfer({
+  // ...
+- network: new StacksTestnet(),
++ network: STACKS_TESTNET,
+});
+```
+
+> [!NOTE]
+> String literal network names are still supported.
+
+```diff
+const transaction = await makeSTXTokenTransfer({
+  // ...
+- network: new StacksTestnet(),
++ network: 'testnet',
+});
+```
+
+> [!NOTE]
+> Custom URLs and fetch functions are still supported via the `api` parameter.
+
+```diff
+const transaction = await makeSTXTokenTransfer({
+  // ...
+- network: new StacksTestnet({ url: "mynode-optional.com", fetchFn: myFetch }), // optional options
++ network: STACKS_TESTNET,
++ api: { url: "mynode-optional.com", fetch: myFetch } // optional params
+});
+```
+
+### Clarity Representation
+
+The `ClarityType` enum was replaced by a readable version.
+The previous (wire format compatible) enum is still available as `ClarityWireType`.
+These types are considered somewhat internal and shouldn't cause breaking changes for most use-cases.
+
+The property holding the value of the data type is now called `value` in all cases.
+Previously, there was a mix of `value`, `list`, `buffer` etc.
+For `bigint` values, the type of the `value` property is a now `string`, for better serialization compatibility.
```

Comment:
Is it to avoid this
```
// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#use_within_json
// @ts-ignore
BigInt.prototype.toJSON = function () {
  return this.toString();
};
```

Are string better than bigint (apart from the bigint serialization issue?)

Reply:
Other than serialization, no there's no downside of bigints that I can think of. Although some environments may not have support for it (not many left, mostly older bundling setups)

Reply:
Ok ok I see. I'm marking this a resolved